lib/charted.js

```diff
@@ -9,16 +9,19 @@ module.exports = function(app, charted_root) {
   if (charted_root === undefined) { charted_root = '/charted/' }
 
   app.use(charted_root + 'get', function (req, res, next) {
-    var get_url = url.parse(req.url, true).query.url
+    var urlObj = url.parse(req.url, true)
 
-    if (!get_url) {
+    if (!urlObj.query.url) {
       res.statusCode = 400
       res.end('Bad Request: no url provided')
-      next()
       return
     }
 
-    request(decodeURIComponent(get_url), function (err, resp, body) {
+    var get_url = urlObj.query.url.match(/:\/\//) ? urlObj.query.url :
+        ('http://' + req.headers.host + '/' +
+         urlObj.query.url.replace(/^\//, ''))
```

Comment:
Well, if it's a relative path then url.Obj.hostname will be `null` (or empty), right? My example above handles relative part, no?

To make sure I'm not misunderstanding what you're trying to solve: sometimes you get relative URLs and you want to make them absolute by appending http://<hostname> in front? If this is the case, you can definitely use `url.format` and `url.parse` and you don't need regular expressions. We can chat about this next week, if you want.